walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,119 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> conflictingTimes = findConflictingTimes(events, request.getAttendees());
+    List<TimeRange> possibleTimes = findPossibleTimes(conflictingTimes, request.getDuration());
+
+    return possibleTimes;
+  }
+
+  public List<TimeRange> findConflictingTimes(Collection<Event> events,
+      Collection<String> attendees) {
+    List<TimeRange> conflicts = findConflicts(events, attendees);
+    List<TimeRange> conflictingTimes = mergeOverlappingRanges(conflicts);
+
+    return conflictingTimes;
+  }
+
+  public List<TimeRange> findConflicts(Collection<Event> events,
```

Comment:
I might also look at adding javadocs for these methods, especially public ones. This and the above method are named similarly and also take a similar set of parameters.